Makefile

```diff
@@ -76,12 +76,15 @@ $(OUTPUT_DIRNAME)/$(DOC_FILENAME).html: $(DOC_FILES) $(FIGURE_FILES)
 	ls -sh $(shell readlink -f $@)
 endif
 
-validate-examples:
+validate-examples: schema/fs.go
 	go test -run TestValidate ./schema
 
-schema-fs:
+schema/fs.go: $(wildcard schema/*.json) schema/gen.go
+	cd schema && printf "%s\n\n%s\n" "$$(cat ../.header)" "$$(go generate)" > fs.go
+
+.PHONY: schema-fs
```

Comment:
Quite a few `.PHONY` entries are collected at the bottom of the Makefile if you want to put this entry there.  On the other hand, there are still a few single-entry `.PHONY`s scattered throughout the Makefile, so there is precedent if you want to keep it here.